File: mobile/lib/core/utils/money.dart
import 'package:intl/intl.dart';

/// Lakh/crore-notation money display ("₹ 1.04Cr"). Previously this exact
/// function -- byte-for-byte -- was copy-pasted as a private `_lakh` in four
/// separate dashboard screens, and none of the four rolled over past lakh,
/// so a 10+ crore portfolio rendered as e.g. "1039.39L" -- a number an
/// Indian collections owner reads as simply wrong.
String lakh(num? v) {
  if (v == null) return '—';
  final cr = v / 10000000;
  if (cr.abs() >= 1) return '${cr.toStringAsFixed(2)}Cr';
  final l = v / 100000;
  if (l.abs() >= 0.01) return '${l.toStringAsFixed(2)}L';
  return NumberFormat.decimalPattern('en_IN').format(v);
}

final _rupeeFormat = NumberFormat.currency(locale: 'en_IN', symbol: '₹', decimalDigits: 0);

/// Plain ₹-prefixed, comma-grouped display for an individual figure (a due
/// amount, a single payment, an EMI) -- as opposed to [lakh]'s aggregate/
/// portfolio notation. Previously this exact `NumberFormat.currency(...)`
/// instance was copy-pasted as a private `_rupee` in five separate screens.
String rupees(num? v) => v == null ? '—' : _rupeeFormat.format(v);

File: frontend/src/utils/money.ts
const rupeeFormatter = new Intl.NumberFormat("en-IN", {
  style: "currency",
  currency: "INR",
  maximumFractionDigits: 0,
});

/**
 * Plain rupee display for an individual figure -- a due amount, a single
 * payment, an EMI. Always carries the ₹ symbol and en-IN comma grouping.
 * Previously this exact shape (bare `Number(v).toLocaleString("en-IN")`,
 * no currency symbol) was copy-pasted as a local `fmtAmount` in eight
 * different files, so a due amount rendered as a bare number indistinguishable
 * from a count.
 *
 * For portfolio/aggregate totals where lakh/crore notation reads better
 * (a dashboard total, a branch rollup), use `lakh()` from
 * `components/dashboard/format.ts` instead -- this function intentionally
 * never switches to L/Cr notation, since an individual customer's ₹18,000
 * due amount read as "0.18L" is worse, not better.
 */
export function rupees(value: string | number | null | undefined): string {
  if (value === null || value === undefined || value === "") return "—";
  const n = Number(value);
  if (Number.isNaN(n)) return "—";
  return rupeeFormatter.format(n);
}
